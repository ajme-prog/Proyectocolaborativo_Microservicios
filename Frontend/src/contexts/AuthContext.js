import React, { useContext, useState, useEffect } from "react";
import { useCookies } from "react-cookie";

const url_api = "http://localhost:3001"
const AuthContext = React.createContext();

export function useAuth() {
  return useContext(AuthContext);
}

export function AuthProvider({ children }) {
  const [cookies, setCookie] = useCookies(["usuario"]);
  const [currentUser, setCurrentUser] = useState();
  const [loading, setLoading] = useState(true);

  async function registrarUsuario(usuario) {
    return fetch(`${url_api}/registro`, {
      method: "PUT",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
      },
      body: JSON.stringify(usuario),
    });
  }

  async function loginUsuario(correo, pwd){
    return fetch(`${url_api}/login`, {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        'Access-Control-Allow-Origin': '*',
      },
      body: JSON.stringify({
        correo: correo,
        pwd: pwd
      }),
    });
  }

  useEffect(() => {
    /*const unsubscribe = auth.onAuthStateChanged((user) => {
      setCurrentUser(user);
      if (user) {
        firestore
          .collection("usuarios")
          .doc(user.uid)
          .get()
          .then((doc) => {
            setUsuarioA(doc.data());
          })
          .then(() => {
            setLoading(false);
          });
      } else {
        setLoading(false);
      }
    });
    return unsubscribe;*/
    async function recuperarAuth () {
        let accessToken = await cookies.accessToken
        let usuarioActual = localStorage.getItem("usuario")

        if(usuarioActual && accessToken){
            usuarioActual = await JSON.parse(usuarioActual)
        }
        setCurrentUser(usuarioActual)
        setLoading(false)
    }

    recuperarAuth()
  }, [cookies]);

  const value = {
    currentUser,
    setCurrentUser,
    cookies,
    setCookie,
    loginUsuario,
    registrarUsuario
  };

  return (
    <AuthContext.Provider value={value}>
      {!loading && children}
    </AuthContext.Provider>
  );
}
