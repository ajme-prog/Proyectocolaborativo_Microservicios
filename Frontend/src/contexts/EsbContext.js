import React, { useContext, useState, useEffect } from "react";

//const grupo1 = "http://35.230.87.64:4091"
const grupo1 = "http://localhost:3001"
const grupo2 = "http://grupo2:4091"
const grupo3 = "http://grupo3:4091"
const grupo4 = "http://grupo4:4091"

const EsbContext = React.createContext();

export function useEsb() {
  return useContext(EsbContext);
}

export function EsbProvider({ children }) {
  
  const [esb, setEsb] = useState();
  
  async function setearESB(esb){
    switch(parseInt(esb)){
      case 1:
        setEsb(grupo1)
        return grupo1
      case 2:
        setEsb(grupo2)
        return grupo2
      case 3:
        setEsb(grupo3)
        return grupo3
      case 4:
        setEsb(grupo4)
        return grupo4
    }
  }

  const value = {
    esb,
    setEsb,
    setearESB
  };

  return (
    <EsbContext.Provider value={value}>
      {children}
    </EsbContext.Provider>
  );
}
