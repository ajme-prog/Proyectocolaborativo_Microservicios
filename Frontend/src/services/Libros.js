const url_api_genero = "http://localhost:4000"
const url_api_libros = "http://localhost:4040"


export async function getGeneros() {
  return fetch(url_api_genero + "/generoliterario", {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    }    
  });
}

export async function Crearlibro(nombre,generos,stock,autor,editorial,id_editorial,paginas,fechapublicacion,idioma,precio,fotografias) {
  let file = fotografias.item(0);
  let foto_base64 = await pFileReader(file);
  foto_base64 = foto_base64.split(",")[1];
  return fetch(url_api_libros+"/libros",{
    method: "POST",
        headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      } ,
      body: JSON.stringify({
        nombre: nombre,
        generos:generos,
        stock:stock,
        autor:autor,
        editorial: editorial,
        id_editorial: id_editorial,
        paginas: paginas,
        fechapublicacion: fechapublicacion,
        idioma: idioma,
        precio: precio,
        foto: foto_base64

      }),
    });
}


export async function DeleteLibro(id_libro) {
  return fetch(url_api_libros+"/libros",{
    method: "DELETE",
        headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      } ,
      body: JSON.stringify({
        id_libro:id_libro
      }),
    });
  
  }
  

function pFileReader(file) {
  return new Promise((resolve, reject) => {
    var fr = new FileReader();
    fr.onload = () => {
      resolve(fr.result);
    };
    fr.readAsDataURL(file);
  });
}


export async function getLibros(id_editorial) {
  return fetch(url_api_libros + "/libros/"+id_editorial, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    }    
  });
}

export async function getunLibro(id_libro) {
  return fetch(url_api_libros + "/libros/unlibro/"+id_libro, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    }    
  });
}

export async function ModificarLibro(id_libro,nombre,generos,stock,autor,editorial,id_editorial,paginas,fechapublicacion,idioma,precio,fotografia) {
  
  let foto_base64 = "";

  if(fotografia){
    let file = fotografia.item(0);
    foto_base64 = await pFileReader(file);
    foto_base64 = foto_base64.split(",")[1];
    console.log(foto_base64);
  }

  return fetch(url_api_libros+"/libros/modificar",{
    method: "POST",
        headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      } ,
      body: JSON.stringify({
        id_libro:id_libro,
        nombre: nombre,
        generos:generos,
        stock:stock,
        autor:autor,
        editorial: editorial,
        id_editorial: id_editorial,
        paginas: paginas,
        fechapublicacion: fechapublicacion,
        idioma: idioma,
        precio: precio,
        foto: foto_base64

      }),
    });
}

