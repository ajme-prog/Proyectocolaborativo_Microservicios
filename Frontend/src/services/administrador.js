const url_api = "http://localhost:3001/administrador"


async function editorialesPendientes(token) {
  console.log(token)
  return fetch(`${url_api}/editoriales-pendientes`, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
      "Authorization": `Bearer ${token}`
    }
  });
}

async function aprobarEditorial(usuario, token){
  return fetch(`${url_api}/aprobar`, {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
      "Authorization": `Bearer ${token}`
    },
    body: JSON.stringify({
      editorial: usuario,
    }),
  });
}

module.exports = { editorialesPendientes, aprobarEditorial }