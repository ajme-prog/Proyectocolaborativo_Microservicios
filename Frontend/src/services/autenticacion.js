const url_api = "http://localhost:3001"

async function registrarUsuario(usuario) {
  return fetch(`${url_api}/registro`, {
    method: "PUT",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
    },
    body: JSON.stringify(usuario),
  });
}

async function loginUsuario(correo, pwd){
  return fetch(`${url_api}/login`, {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
    },
    body: JSON.stringify({
      correo: correo,
      pwd: pwd
    }),
  });
}

export{ registrarUsuario, loginUsuario };
