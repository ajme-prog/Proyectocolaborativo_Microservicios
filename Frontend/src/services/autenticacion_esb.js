async function registrarUsuario(usuario, esb) {
  return fetch(`${esb}/auth/registro`, {
    method: "PUT",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
    },
    body: JSON.stringify(usuario),
  });
}

async function loginUsuario(correo, pwd, esb){
  console.log("esb", esb)
  
  return fetch(`${esb}/auth/login`, {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
    },
    body: JSON.stringify({
      email: correo,
      passwd: pwd
    }),
  });
}

export{ registrarUsuario, loginUsuario };
