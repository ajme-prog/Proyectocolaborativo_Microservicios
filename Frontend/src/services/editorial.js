const url_api = "http://localhost:9010/impuestos"


async function recuperarImpuestos() {
  return fetch(`${url_api}/get_countries`, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
      'Access-Control-Allow-Origin': '*',
    }
  });
}

export { recuperarImpuestos }