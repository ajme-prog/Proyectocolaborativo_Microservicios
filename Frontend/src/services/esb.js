const grupo1 = "http://35.230.87.64:4091";
const grupo2 = "http://35.232.176.148:4091";
const grupo3 = "http://35.222.108.70:4091";
const grupo4 = "http://34.68.53.63:4091";

export async function setearESB(esb) {
  switch (parseInt(esb)) {
    case 1:
      localStorage.setItem("esb", JSON.stringify(grupo1));
      return grupo1;
    case 2:
      localStorage.setItem("esb", JSON.stringify(grupo2));
      return grupo2;
    case 3:
      localStorage.setItem("esb", JSON.stringify(grupo3));
      return grupo3;
    case 4:
      localStorage.setItem("esb", JSON.stringify(grupo4));
      return grupo4;
  }
}

export async function getESB() {
  return await JSON.parse(localStorage.getItem("esb"));
}
