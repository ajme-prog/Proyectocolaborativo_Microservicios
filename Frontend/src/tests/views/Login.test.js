import React from 'react';
import { mount, shallow } from 'enzyme';
import Login from 'views/auth/Login';
import { useAuth } from 'contexts/AuthContext';
import { loginUsuario } from 'services/autenticacion';
import { MemoryRouter } from 'react-router-dom';
jest.mock('services/autenticacion');
jest.mock('contexts/AuthContext');

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));


describe('Pruebas en <Login />', () => {



    test('debe de mostrarse correctamente', () => {
        useAuth.mockReturnValue({
            setCookie: jest.fn(),
            setCurrentUser: jest.fn()
        })
        const wrapper = shallow(<Login />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de realizar el dispatch y hacer la navegaciÃ³n', () => {
        useAuth.mockReturnValue({
            setCookie: jest.fn(),
            setCurrentUser: jest.fn()
        });

        loginUsuario.mockReturnValue({
            status: 200,
            accessToken: '123sfafa32',
            usuario: {
                nombre: 'Ronald',
                tipo: 0 //admin
            }
        })
        const wrapper = mount(
            <MemoryRouter>
                <Login />
            </MemoryRouter>
        );
        wrapper.find('button').prop('onClick');
        // handleSubmit();
        // console.log(wrapper.find('button').simulate('click'));


        //wrapper.find('button').prop('onClick');
        expect(wrapper.find('Router').prop('history').location.pathname.trim()).toBe('/');

        //expect(mockHistoryPush).toHaveBeenCalledWith('/admin/settings');
    });


})
