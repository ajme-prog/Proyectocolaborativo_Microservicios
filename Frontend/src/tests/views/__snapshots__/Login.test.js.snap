// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Login /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <div
    className="container mx-auto px-4 h-full"
  >
    <div
      className="flex content-center items-center justify-center h-full"
    >
      <div
        className="w-6/12 lg:w-4/12 px-4"
      >
        <div
          className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-lg bg-blueGray-200 border-0"
        >
          <div
            className="rounded-t mb-0 px-6 py-6"
          >
            <div
              className="text-center mb-3"
            >
              <h6
                className="text-blueGray-500 text-sm font-bold"
              >
                BOOKSA
              </h6>
            </div>
            <hr
              className="mt-6 border-b-1 border-blueGray-300"
            />
          </div>
          <div
            className="flex-auto px-4 lg:px-10 py-10 pt-0"
          >
            <div
              className="text-blueGray-400 text-center mb-3 font-bold"
            >
              <small>
                Ingresa mediante tu correo y contraseña
              </small>
            </div>
            <form
              onSubmit={[Function]}
            >
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block uppercase text-blueGray-600 text-xs font-bold mb-2"
                  htmlFor="grid-password"
                >
                  Correo
                </label>
                <input
                  className="border-0 px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  placeholder="Correo electrónico"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="relative w-full mb-3"
              >
                <label
                  className="block uppercase text-blueGray-600 text-xs font-bold mb-2"
                  htmlFor="grid-password"
                >
                  Contraseña
                </label>
                <input
                  className="border-0 px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                  placeholder="Contraseña"
                  required={true}
                  type="password"
                />
              </div>
              <div
                className="text-center mt-6"
              >
                <button
                  className="bg-blueGray-800 text-white active:bg-blueGray-600 text-sm font-bold uppercase px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 w-full ease-linear transition-all duration-150"
                  type="submit"
                >
                  Iniciar sesión
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          className="flex flex-wrap mt-6 relative"
        >
          <div
            className="w-full text-center"
          >
            <Link
              className="text-blueGray-200"
              to="/auth/register"
            >
              <small>
                ¿No tienes una cuenta? Registrarme
              </small>
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
